diffsync/store/__init__.py

```diff
@@ -159,6 +159,24 @@ def get_or_instantiate(
 
         return obj, created
 
+    def get_or_instantiate_model(self, obj: "DiffSyncModel") -> Tuple["DiffSyncModel", bool]:
```

Comment:
Maybe this should be called `get_or_add_model_instance`, seeing as
- the model class instance is already instantiated
- the model itself is already known to the `DiffSync` instance at this point (i.e. we aren't "adding" a model)
- "`get_or_add`" also accurately reflects the methods called against `self`